src/week01/sungHee/p1/p1_SungHee.java

```diff
@@ -0,0 +1,90 @@
+package week01.sungHee.p1;
+
+import java.io.*;
+import java.util.*;
+
+/**
+ * 문제 링크 : https://www.acmicpc.net/problem/2178
+ * 메모리 : 12304 KM
+ * 시간 : 76 ms
+ */
+
+/**
+ * 적용 알고리즘 : BFS(최단 경로를 찾는 문제)
+ * 
+ * miro[y][x]의 값을 depth로 생각
+ * miro[0][0]을 2로 설정하여
+ * miro[y][x]의 값이 1인 경우에만 지나갈 수 있다.
+ * (0이면 못지나가는 칸, 1초과면 이미 지나간 칸)
+ */
+
+public class p1_SungHee {
+
+    static int[][] miro;
+    static int N, M;
+    static boolean[][] visited;
+
+    static int[] dx = {1, 0, -1, 0}; //x방향배열
+    static int[] dy = {0, -1, 0, 1}; //y방향배열
+
+    private static class State {
+        public final int y;
+        public final int x;
+
+        public State(int y, int x) {
+            this.y = y;
+            this.x = x;
+        }
+    }
+
+
+    public static void main(String[] args) throws IOException {
+        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+        StringTokenizer st = new StringTokenizer(br.readLine());
+        
+        N = Integer.parseInt(st.nextToken());
+        M = Integer.parseInt(st.nextToken());
+
+        miro = new int[N][M];
+
+        for (int i = 0; i < N; i++) {
+            String line = br.readLine();
+            for(int j =0; j<M; j++) {
+                miro[i][j] = line.charAt(j) - '0';
+            }
+        }
+
+
+        miro[0][0] = 2;
+
+        bfs(0,0);
+        System.out.println(miro[N-1][M-1] -1);
```

Comment:
제출한 코드는 통과 했는데 miro[0][0] 이 0인경우 통과 못할 것 같네요 😅 
아래처럼 바꿔야 할 것 같아요!
```suggestion
        if (miro[0][0] == 0) {
             System.out.println(0);
             return;
        }
        
        miro[0][0] = 2;

        bfs(0,0);
        System.out.println(miro[N-1][M-1] -1);
```